entity/contents/contents.h

```diff
@@ -33,6 +34,14 @@ class Contents {
                       const Entity& entity,
                       RenderPass& pass) const = 0;
 
+  /// @brief Returns true if this Contents is a FilterContents. This is useful
+  ///        for downcasting to FilterContents without RTTI.
+  virtual bool IsFilter() const;
```

Comment:
Yeah, this is a knee jerk thing I've done sometimes in my personal projects. I agree that there's a much better approach here. Since _all_ contents can actually render to a texture now, I went ahead and moved `RenderToTexture` into Contents itself.

We'll most likely need this in a number of non-filter situations down the line anyways. Now that I think about it, that's probably the real star of the show in this PR. :)